### ISSUE TITLE
Cant seem to reinstall MO2 after failed MO2 update from MO2 GUI

### ISSUE THREAD

**Issue by tetoNidan** [sonic2kk/steamtinkerlaunch#1021]
## System Information
- SteamTinkerLaunch version: v12.12
- Distribution: Garuda Arch Linux 
- Installation Method: Chaotic AUR

## Issue Description

So I tried to update MO2 through MO2's GUI and it failed. Not sure if it was due to `firejail` or due to me  symbolically linking my MO2 folder to save space on my OS SSD but at least that time I got to the MO2 license agreement stage before failing. Sorry I don't have the those logs though. I could potentially get them though if your interested as I have a backup of the old MO2 folder.

So I have tried many re-installs and have not gotten past the Microsoft Redistributables Installer.

I have tried the earliest versions of proton I have installed. proton 8.0.3c I believe and 7 as well as a few newer GE versions.

I have disabled `firejail` so I know that things are not being sandboxed. 

Not sure what to try next. It appears that MO2 is installed as the folder and files seem to be all there. Just not starting so I'm thinking it's a proton issue but I really don't know, just a shot in the dark. What version of proton do you recommend for install of the new MO2? I had changed the default quite some time ago and don't likely have it installed any more.

Maybe these logs will help point me in the right direction.

I have a backup of the old MO2 folder so if all else fails I can likely re symlink it to get it going again.

Hope things are going well sonic2kk and as always thanks! 

## Logs

# installMO2_protonrun.log 

Proton: Upgrading prefix from None to 8.0-103 (/home/tool/.config/steamtinkerlaunch/mo2/compatdata/)
fsync: up and running.
wine: Using setpriority to control niceness in the [-11,11] range
steam.sh[27048]: Running Steam on garuda Soaring 64-bit
steam.sh[27048]: STEAM_RUNTIME is enabled automatically
setup.sh[27506]: Updating Steam runtime environment...

(zenity:27652): Gtk-WARNING **: 23:47:24.122: Theme parser error: gtk-dark.css:2588:14-18: Negative values are not allowed

(zenity:27652): Adwaita-WARNING **: 23:47:24.142: Using GtkSettings:gtk-application-prefer-dark-theme with libadwaita is unsupported. Please use AdwStyleManager:color-scheme instead.
setup.sh[27506]: Found newer runtime version for 32-bit libGLU.so.1. Host: 1.3.1 Runtime: 1.3.8004
setup.sh[27506]: Forced use of runtime version for 32-bit libcurl-gnutls.so.4
setup.sh[27506]: Forced use of runtime version for 32-bit libcurl.so.4
setup.sh[27506]: Forced use of runtime version for 32-bit libdbusmenu-glib.so.4
setup.sh[27506]: Forced use of runtime version for 32-bit libdbusmenu-gtk.so.4
setup.sh[27506]: Forced use of runtime version for 32-bit libgtk-x11-2.0.so.0
setup.sh[27506]: Found newer runtime version for 64-bit libGLU.so.1. Host: 1.3.1 Runtime: 1.3.8004
setup.sh[27506]: Forced use of runtime version for 64-bit libcurl-gnutls.so.4
setup.sh[27506]: Forced use of runtime version for 64-bit libcurl.so.4
setup.sh[27506]: Found newer runtime version for 64-bit libdbusmenu-glib.so.4. Host: 4.0.12 Runtime: 4.0.13
setup.sh[27506]: Found newer runtime version for 64-bit libdbusmenu-gtk.so.4. Host: 4.0.12 Runtime: 4.0.13
steam.sh[27048]: Steam client's requirements are satisfied

# startMO2_.log

wineserver: using server-side synchronization.
wine: Using setpriority to control niceness in the [-11,11] range
creating minidump for the current process
trying file '.\ModOrganizer-2.5.1rc2-20240120T064813.dmp'
writing mini minidump
minidump written correctly

# steamtinkerlaunch.log

https://pastebin.com/JV0DYzUL

**Comment by tetoNidan** [comment]
So I just read the part about mo2 v 2.5+ not being supported my proton in the mo2 install page and I assume that is the issue I'm having. So Ill down grade the install till wine/proton catches up. I'll close this.

As always thanks sonic2kk!

**Comment by tetoNidan** [comment]
So I tried downgrading mo2 by first removing `.config/steamtinkerlaunch/mo2` then running `steamtinkerlaunch mo2 install ~/Downloads/Mod\ Organizer\ 2-6194-2-4-4-1640622655.exe` version 2.4.4 from nexusmods, STL keeps downloading the latest version Mod.Organizer-2.5.1rc2.exe and installing that version instead. I have removed the 2.5.1 from `.config/steamtinkerlaunch/downloads/mo2` and tried it again but STL keeps re-downloading 2.5+ to install. I have also passed an absolute path to the `install` command.

I looked for an option to set custom mo2 executable `USEMO2CUSTOMINSTALLER` but cannot find it? I opened up the `global.conf` thinking it might be in there but it was not. Is that the correct config it would be in?

Chaotic AUR version installed or STL is 12.12. Is the AUR version somehow not up to date with current STL version even though it shows the same version?

**Comment by suamor** [comment]
I recommend not to spend too much time for the STL configuration. Make a backup from the steamuser folder (MO2 pfx folder) and - if changed - regedit settings. then rename or delete the whole STL prefix and reinstall MO2.
Otherwise you could build proton/wine-9.0 tkg yourself or download a tkg daily build.

**Comment by tetoNidan** [comment]
I just uninstalled STL completely and built from 12.12 git source hoping something was wrong with the AUR package not letting me install older version of MO2 using `steamtinkerlaunch mo2 install /path/to/file`. I even downloaded MO2 2.4.4 directly from MO2's release page thinking that there might be some pattern matching getting in the way as the filename from nexusmods is a bit different. No luck though. Still cant downgrade, STL only initiates download and install of current MO2 release only.

I have managed to get back up and running after reinstalling from the AUR and restoring a backup of MO2 as well as a backup of `.config/steamtinkerlaunch`. 

If you guys want me to open a new bug report for this specific issue I can do that and we could close this one as it was initially operator error. I didn't read the docs in a while and was updating based on a conversation I had a while ago with sonic2kk on another issue.

As always thanks STL team!

**Comment by suamor** [comment]
I have organized my MO2 installation like this:
- Make an installation with MO2 24
- Rename folder to f.e. C:/Modding/MO2-24. (on Linux C:/Moddding would be f.e. /media/user/extern/Modding)
- Make an installation with MO2 25
- Rename folder to f.e. C:/Modding/MO2-25.
- Copy or move your mods folder back to C:/Modding
- Copy or move your overwrite folder back to C:/Modding
- Copy or move your mod archive folder back to C:/Modding
- Copy your profiles folder back to C:/Modding/MO2-24 and to C:/Modding/MO2-25
- Link your mods folder to C:/Modding/mods (f.e. cd MO2-24; ln -s /media/user/extern/Modding/mods ./mods)
- Link your overwrite folder to C:/Modding/mods (f.e. cd MO2-24; ln -s /media/user/extern/Modding/overwrite ./overwrite)
- Link your overwrite folder to C:/Modding/mods (f.e. cd MO2-25; ln -s /media/user/extern/Modding/overwrite ./overwrite)
- Link your MO2-24 or MO2-25 to MO2 (f.e. cd /media/extern/Modding; ln -s /media/user/extern/Modding/MO2-24 ./MO2)

Now you have to separate installations with a common overwrite and mods folder. profiles are separate, so you may need from now on to change settings in both MO2-24 and MO2-25. Finally remove MO2-24 when MO2-25 runs stable.

**Comment by sonic2kk** [comment]
Hi there @tetoNidan, it sounds like you're using an outdated version of SteamTinkerLaunch as you seem to have gathered. The options you're listing are NOT available in v12.12, the wiki and myself DO NOT accommodate the stable releases of STL anymore.

Issues using STL with MO2 v2.5+ have been reported [many, many times now](https://github.com/sonic2kk/steamtinkerlaunch/issues/1018#issuecomment-1893036240) too.

I recommend doing the following:
- Updating to STL-git.
- Removing your ModOrganizer 2 install (backing up any mods directories in the process).
    - I cannot remember exactly where mods are installed at this time, but if you load up a game that you used MO2 v2.4.4 with that was not updated, or use `steamtinkerlaunch mo2 start` which also should not have updated (unless you updated that too by mistake), then you can view the paths in the MO2 settings under the Paths tab.
    - For me, my mods for Oblivion are installed to the following path: `Z:/home/emma/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/users/steamuser/AppData/Local/ModOrganizer/Oblivion/mods` -- but this *may be different* for you, so please be careful and double-check first!! :-)
- Removing any updated MO2 installations from the game prefix (MO2 is at `/path/to/steamapps/compatdata/<appid>/pfx/drive_c/Modding` iirc).
- Re-install ModOrganizer 2 v2.4.4 with `steamtinkerlaunch mo2 install` (STL git should be pinned to this)
- Copy your mods folder back into the ModOrganizer 2 install at `$HOME/.config/steamtinkerlaunch/mo2`.

In the future, please do not update MO2, or any software under Wine, as it is likely to fail. However, as @suamor has pointed out, a build of Proton-tkg based on Wine 9 (or really, Wine 8.21 or above, as that's when the path for the Qt6 feature was introduced) should also work. Game compatibility *may* vary though, as you're using a much more bleeding edge version of Wine! You will likely also need to re-install anyway with this version of Proton, as installation of MO2 2.5+ may not have succeeded with other Proton versions.

<hr>

I hope this helps. I am not sure exactly what the steps the user above is describing are about, they may also help, but really the main solution here is to backup your mods, remove your MO2 install(s), and then re-install with STL v12.12. Sticking with stable STL is a horrible idea that, by not releasing updates, I had hoped users would simply stop doing. That hasn't worked, and I fear that after releasing v14.0 the problem will just come around again :frowning_face: 

There should be an AUR package available for STL-git, but you can also remove the AUR package entirely and install by cloning the repo and installing with `sudo make install`. If you ever want to go back to the AUR package or install another way, from any STL cloned repo you can run `sudo make uninstall`. You can check what these all do in the [Makefile](https://github.com/sonic2kk/steamtinkerlaunch/blob/6fc2a35a9661a2ecbeb6820a9573f769927bbde8/Makefile).

Thanks!

**Comment by tetoNidan** [comment]
Hi [sonic2kk](https://github.com/sonic2kk),

> Issues using STL with MO2 v2.5+ have been reported https://github.com/sonic2kk/steamtinkerlaunch/issues/1018#issuecomment-1893036240 too.

Yeah I figured that out a bit too late. I was recalling our previous conversation where we were discussing updating and you said that I might be able to just update through the MO2 app, you said that you've never tried it because you never had to. Only when it didn't work out did I come here to read the wiki and file a bug report.

I am using STL 12.12-1.2 from the AUR. I have also installed and tested the latest [git](url), the most current release is Gates of Steiner? Or is there an experimental branch that I should be looking for or are you talking about downloading the source [`master branch`] for the repository and not from the releases page? 

In any event I have successfully restored my mo2 install using a backup with a little debugging. As you pointed out 12.12 is no longer current so I was assuming that the wiki instructions were for 12.12 as well. Since I have a working install and it's not a bug as I'm using an old version without features I was trying to use there is no reason to keep this open. I will take note that the wiki does not necessarily pertain to the current stable release and likely refers to the `master` branch. 

As always thanks for your work sonic2kk!

**Comment by sonic2kk** [comment]
> or are you talking about downloading the source [master branch] for the repository and not from the releases page?

Yes, don't use the releases. Even when v14.0 is out, please don't use a stable version as you won't get the latest fixes :-) There may also be an AUR package for STL-git, there was a while ago but I haven't checked recently.

I don't know if release frequency will pick up after v14.0, I am considering yearly releases similar to what Winetricks does. I would rather not limit users to waiting on a release though, fixes are super accessible once available with STL since it's just a Bash script.

If I didn't think it would result in constant duplicate issues, I would turn off releases altogether.

> I will take note that the wiki does not necessarily pertain to the current stable release and likely refers to the master branch.

Yup! The wiki and all information about the project will always point to the latest version, as releases are not really "stable"s so much as development checkpoints. Fixes are not backported, for example, and if a user is not using the latest commit I will ask them to check it as the first step when opening an issue (the issue template also encourages checking the latest commit too).

Since STL is just a bash script that needs no compilation, it should be straightforward to update it :-)

Glad everything is working for you regardless of all of this. However please I encourage you to not stick to the releases, honestly with most software you're probably better off with the latest version you can get your hands on! 

But it's working for you and that's what matters. Happy gaming!

**Comment by tetoNidan** [comment]
> There may also be an AUR package for STL-git, there was a while ago but I haven't checked recently.

I usually stick to the AUR version whether from the Chaotic repo or the AUR directly. The AUR version is 12.12 and that is what the Chaotic AUR uses to compile from. I don't think they will update the package till the STL version # changes so its good to know that they will likely be a year behind. It looks like the stl-git version is on 11.11 last updated in late 2022 just an FYI.

I just like the ease of updating/removing all packages through pacman/yay as I don't have to keep or redownload anything for reinstallation or removal but I might make an exception for STL, it's not that hard to do for one program, it's when it's 5+ that it gets a little cumbersome.

Any way it's always nice to chat, hope you had a wonderful holiday season and you and your loved ones are in good health! Thanks sonci2kk.

**Comment by sonic2kk** [comment]
> It looks like the stl-git version is on 11.11 last updated in late 2022 just an FYI.

I think this is just a "cosmetic" version, since the [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=steamtinkerlaunch-git) should just pull from GitHub rather than any specific release.

> I just like the ease of updating/removing all packages through pacman/yay

Yeah, I do too, and usually I will use this to get `-git` versions of packages. For example, `lutris-git`, `gimp-devel`, previously I used Mesa-git as well from Chaotic-AUR (encountered a GPU hang and had to rollback, but will probably go to Mesa-git again soon). But I agree that using a Makefile is usually a last resort. [The previous maintainer agreed](https://github.com/sonic2kk/steamtinkerlaunch/issues/483#issuecomment-1147080730), but to make this a lot more straightforward, I added an `uninstall` block to the Makefile (#608). Many projects when installed from source don't have an easy uninstall path, but I wanted to make this as straightforward as possible.

There have also been improvements to using "local" installs of SteamTinkerLaunch, based on the information given in #548 and improvements since then (can't find all the PRs, but most of the work was done in v11.11 onwards). Essentially you just put STL somewhere on your harddrive, add the path to the script to your `PATH`, and run `steamtinkerlaunch compat add`. STL will then be able to detect how it is running and use the files in the local folder. This is how STL works on SteamOS and [also how it works when installed from ProtonUp-Qt](https://github.com/DavidoTek/ProtonUp-Qt/blob/main/pupgui2/resources/ctmods/ctmod_steamtinkerlaunch.py#L226-L435).

The downside to a "local" install is that you have to add STL to your path manually (although ProtonUp-Qt can do it), but mainly that you don't get your dependencies managed for you. ProtonUp-Qt does list the hard dependencies though, and the wiki has a [full list of dependencies](https://github.com/sonic2kk/steamtinkerlaunch/wiki/Installation#dependencies-1).

However, you do still make a good point about package managers. A few projects package STL out of the box now (Bazzite, winesapOS, Nobara) and they stick to the stables. So that does put me off a yearly release "cycle". Nothing is really set in stone yet, I'll see how I feel after v14.0 comes out and take things from there :-) I am tentatively hoping to release v14.0 in a few months (see #992 for roadmap).

> Any way it's always nice to chat, hope you had a wonderful holiday season and you and your loved ones are in good health! 

I appreciate that, you too! I have been taking a mental health break from public projects and just checked in today, but I will likely continue my break for a little bit longer. Take care!

**Comment by tetoNidan** [comment]
**This is not an issue that needs any direct attention.**

So I'm testing out an update to MO2 2.5.2. I had an old `.config/steamtinkerlaunch/mo2` backup from when I installed STL-12.12, yeah I know I should switch to git, chaoticAUR is just easier. 

Any way, I needed to rebuild from scratch because my load order was fubar from years of mod-neglect but I wanted the latest MO2 before I started. It seems that the MO2 updater works just fine from inside MO2. I haven't began modding yet so I actually don't know for sure if everything is working as intended. But I'm gonna find out.

OS: Garuda - Arch'ish
MO2 Version  :  2.5.2
Wine Version : wine-9.21 (Staging)
STL Version:   : 12.12

Thanks for doing what you do man! Might take me a few days but I'll report back on my progress.

**Comment by tetoNidan** [comment]
So after a bit of trial and error I came to the conclusion that I probably had to install vcredist2022 to the skyrim prefix. Not sure if it was installed in the MO2 prefix when I updated MO2 through the GUI. Might install it to the prefix just to be sure. There are a couple of new mods that are requiring the latest version like AddressLibrary. I think it's even a requirement of MO2. I did install the dotnet48. ~~MO2 would loose skyrims load order after launching the game.~~ After further tinkering and research I found out that the mod [EngineFixes](https://forums.nexusmods.com/topic/13458736-a-fix-for-all-plugins-disabling-after-161130/) was having an issue and the toml file needed to be edited to enable mods again. 

All is good now with vcredist2022 installed though.

Everything seems to work. All except the loot. loot is not displaying it's return. If I remember right the old MO2 prefix I was using for testing I had already broken loot. I find it really easy to break loot. I think It might be using mismatched versions of proton. This is just a guess though as my unfamiliarity with the inner working of proton/wine, I might be seeing correlation not causation.

Any way I think it's pretty neat that the updater works from inside MO2. Might need to automate a bash script to preinstall whatever dependencies MO2 needs before update maybe? Maybe not?

I know my STL version is quite outdated but I hope this helps a bit. As stated above this does not require your intervention or help, just me tinkering. If you have any questions I'd be happy to answer.  As always thank you sonic2kk!

**Comment by sonic2kk** [comment]
`vcredist2022` was not historically required for MO2. Perhaps it is needed for Skyrim, though, or a Skyrim mod / modloader?

Just wanted to clarify that vcredist2022 was not required for ModOrganizer 2 as of some months back (whenever Wine got the accent colour patch to allow MO2 2.5+ to work). If this is a new requirement it may be something we'll need to look into!

> Any way I think it's pretty neat that the updater works from inside MO2. Might need to automate a bash script to preinstall whatever dependencies MO2 needs before update maybe? Maybe not?

it is good that the update worked, if MO2 does have new dependencies needed to run under Wine, ideally we would be able to handle those. `vcredist2022` should be lightweight enough to install ourselves if we need to :-)

<hr>

As for the issues with LOOT, some users have reported problems with LOOT under Wine, but I have not encountered such issues. Then again, it _has_ been a long time since I've used any mod manager (I avoid modding games these days). But if anyone figures out that there is a Wine-specific fix or if SteamTinkerLaunch is doing something wrong, we can look into it!

I do recall there was a bug that was fixed a while back around how we set ModOrganizer 2 paths, it seems I had horribly broken it. If I remember this was mainly diagnosed with Fallout New Vegas, but it could also apply to Skyrim (believe it was diagnosed in #1130, and fixed with 144347b1c1b322bda8d3cdc5559ad284a9f8578c).

<hr>

Thanks for the update nonetheless!
